File: Maximum Sum Circular Subarray.cpp
class Solution {
public:
    int kadane(vector<int> A)
    {
        int answer=A[0],sum=0;
        for(int x:A)
        {
            sum=max(x,sum+x);
            answer=max(answer,sum);
        }
      return answer;  
    }
    int maxSubarraySumCircular(vector<int>& A) {
        int wrap=0;
        int ans1=kadane(A);
        for(int i=0;i<A.size();i++ )
        {
            wrap+=A[i];
            A[i]=-A[i];
        }
        int ans2= wrap+kadane(A);
        if (ans2==0)
            return ans1;
        return ans1>ans2?ans1:ans2;
    }
    
};

File: Golang Solutions/Maximum Sum Circular Subarray.go
func kadane(A[] int) int{
    answer:=A[0]
    sum:=0
    for _,x := range A{
        if x>sum+x{
            sum=x;
        }else{
            sum+=x
        }
        if answer<sum{
            answer=sum
        }
    }
    return answer
}
func maxSubarraySumCircular(A []int) int {
    wrap:=0
    ans1:=kadane(A)
    for i,x := range A{
        wrap+=x
        A[i]=-A[i]
    }
    ans2:=wrap+kadane(A)
    if ans2==0 || ans1>ans2{
        return ans1
    }else{
        return ans2
    }
}
   